rqt_py_common/src/rqt_py_common/message_field_type_helpers.py

```diff
@@ -0,0 +1,407 @@
+
+      # String weirdness
+      string<=5       -> 'string<5>'                -> str
+      string[3]       -> 'string[3]'                -> List[str]
+      string[]        -> 'sequence<string>'         -> List[str]
+      string<=5[3]    -> 'string<5>[3]'             -> List[str]
+      string<=5[<=10] -> 'sequence<string<5>, 10>'  -> List[str]
+      string<=5[]     -> 'sequence<string<5>>'      -> List[str]
+      string[<=10]    -> 'sequence<string, 10>'     -> List[str]
+
+    :param field_type: the field type in the form generated by msg.get_fields_and_field_types
+    :type field_type: str
+    '''
+
+    def __init__(self, field_type: str):
+        super(MessageFieldTypeInfo, self).__init__()
+
+        self.base_type_str = base_type_str(field_type)
+        self.is_array = is_array(field_type)
+        self.is_static_array = is_static_array(field_type)
+        self.static_array_size = static_array_size(field_type)
+        self.is_bounded_array = is_bounded_array(field_type)
+        self.bounded_array_size = bounded_array_size(field_type)
+        self.is_unbounded_array = is_unbounded_array(field_type)
+        self.is_bounded_string = is_bounded_string(field_type)
+        self.bounded_string_size = bounded_string_size(field_type)
+
+    def get_field_type_info_as_dict(self) -> Dict[str, Any]:
+        '''Returns the type info as a dictionary'''
+        info_as_dict = {}
+        for slot in MessageFieldTypeInfo.__slots__:
+            info_as_dict[slot] = getattr(self, slot)
+        return info_as_dict
+
+
+__END_OF_BASE_TYPE_DELIM = ['<', '[', ',', '>']
+
+
+def base_type_str(field_type: str) -> str:
+    '''
+    The base type of the field_type (eg, uint8, my_msgs/Custom, ...)
+
+    (Removes the array information leaving just the slot class str.)
+
+    :param field_type: The field_type such as one as created by
+        get_fields_and_field_types()
+
+    :type field_type: str
+
+    :returns: the field_type of with the list component stripped out
+    :rtype str: str
+    '''
+    if field_type.startswith(SEQUENCE_PREFIX):
+        field_type = field_type[SEQUENCE_PREFIX_LEN:]
+
+    for delim in __END_OF_BASE_TYPE_DELIM:
+        delim_ix = field_type.find(delim)
+        if delim_ix >= 0:
+            field_type = field_type[:delim_ix]
+
+    return field_type
+
+
+def is_array(field_type: str) -> bool:
+    '''
+    If the field_type is a static, bounded or unbounded array
+
+    :param field_type: Field type as produced by get_fields_and_field_types()
+    :type field_type: str
+
+    :rtype: bool
+    '''
+    return (field_type.startswith(SEQUENCE_PREFIX) or
+            field_type.find('[') >= 0)
+
+
+def is_static_array(field_type: str) -> bool:
+    '''If the field_type is a static sized array (eg. uint8[4])'''
+    return field_type.find('[') >= 0
+
+
+def static_array_size(field_type) -> int:
+    '''If `is_static_array == True` then `len(msg.field_type)`'''
```

Comment:
Give a more appropriate docstring description and describe the return failure case